src/ln/channelmanager.rs

```diff
@@ -1487,41 +1522,31 @@ impl ChainListener for ChannelManager {
 	}
 }
 
+macro_rules! handle_error {
+	($self: ident, $internal: expr, $channel_id: expr) => {
+		match $internal {
+			Ok(msg) => Ok(msg),
+			Err(e) => {
+				match e.action {
+					Some(msgs::ErrorAction::DisconnectPeer { msg: _ }) => {
+						$self.force_close_channel(&$channel_id);
+					},
+					Some(msgs::ErrorAction::IgnoreError) => {},
+					Some(msgs::ErrorAction::SendErrorMessage { msg: _ }) => {
+						$self.force_close_channel(&$channel_id);
```

Comment:
Do we want a force_close i.e an unilateral close on every error case ? maybe a close_channel here

Reply:
No, I think any case where we hit an ErrorMessage we'll want a full unilateral close, sadly.

That said, we do need to extend this a bit as an ErrorMessage with a channel_id of [0u8; 32] indicates we want to force-close all channels from a given node_id, not just the one.